mat/triangular.go

```diff
@@ -452,15 +452,51 @@ func (t *TriDense) MulTri(a, b Triangular) {
 		defer restore()
 	}
 
-	// TODO(btracey): Improve the set of fast-paths.
+	var aDiag, bDiag bool
+	switch aa := aU.(type) {
```

Comment:
These type switches can be `if aa, ok := aU.(Diagonal); ok {` rather than the switch.